src/pcnt.rs

```diff
@@ -0,0 +1,578 @@
+        let evt_type: pcnt_evt_type_t = PcntEventType::only(evt_type).as_repr();
+        unsafe { esp!(pcnt_event_enable(self.unit, evt_type)) }
+    }
+
+    /// Disable PCNT event of PCNT unit
+    /// 
+    /// @param evt_type Watch point event type.
+    ///                All enabled events share the same interrupt (one interrupt per pulse counter unit).
+    /// returns
+    /// - ()
+    /// - EspError
+    pub fn event_disable(&self, evt_type: PcntEvent) -> Result<(), EspError> {
+        let evt_type: pcnt_evt_type_t = PcntEventType::only(evt_type).as_repr();
+        unsafe { esp!(pcnt_event_disable(self.unit, evt_type)) }
+    }
+
+    fn only_one_event_type(evt_type: PcntEventType) -> Result<pcnt_evt_type_t, EspError> {
+        match evt_type.iter().count() {
+            1 => Ok(evt_type.as_repr()),
+            _ =>Err(EspError::from(ESP_ERR_INVALID_ARG as esp_err_t).unwrap()),
+        }
+    }
+
+    /// Set PCNT event value of PCNT unit
+    /// 
+    /// @param evt_type Watch point event type.
+    ///                All enabled events share the same interrupt (one interrupt per pulse counter unit).
+    /// 
+    /// returns
+    /// - ()
+    /// - EspError
+    pub fn set_event_value(&self, evt_type: PcntEventType, value: i16) -> Result<(), EspError> {
+        let evt_type = Self::only_one_event_type(evt_type)?;
+        unsafe {
+            esp!(pcnt_set_event_value(
+                self.unit, evt_type, value
+            ))
+        }
+    }
+
+    /// Get PCNT event value of PCNT unit
+    /// 
+    /// @param evt_type Watch point event type.
+    ///                All enabled events share the same interrupt (one interrupt per pulse counter unit).
+    /// 
+    /// returns
+    /// - i16
+    /// - EspError
+    pub fn get_event_value(&self, evt_type: PcntEventType) -> Result<i16, EspError> {
+        let evt_type = Self::only_one_event_type(evt_type)?;
+        let mut value = 0i16;
+        unsafe {
+            esp!(pcnt_get_event_value(
+                self.unit,
+                evt_type,
+                &mut value as *mut i16
+            ))?;
+        }
+        Ok(value)
+    }
+
+    /// Get PCNT event status of PCNT unit
+    /// 
+    /// returns
+    /// - i32
+    /// - EspError
+    // TODO: status is a bit field!
+    pub fn get_event_status(&self) -> Result<u32, EspError> {
+        let mut value = 0u32;
+        unsafe {
+            esp!(pcnt_get_event_status(
+                self.unit,
+                &mut value as *mut u32
+            ))?;
+        }
+        Ok(value)
+    }
+
+    /// Configure PCNT pulse signal input pin and control input pin
+    /// 
+    /// @param channel PcntChannel
+    /// @param pulse_io Pulse signal input pin
+    /// @param ctrl_io Control signal input pin
+    /// 
+    /// @note  Set the signal input to PCNT_PIN_NOT_USED if unused.
+    /// 
+    /// returns
+    /// - ()
+    /// - EspError
+    pub fn set_pin<'a>(
```

Comment:
This method is unsafe for the very same reason `channel_config` is unsafe. Remove.

Reply:
removed for now